src/sensors/tests/test_irradiancemeter.py

```diff
@@ -0,0 +1,140 @@
+import math
+import random
+
+import numpy as np
+import pytest
+
+import enoki as ek
+import mitsuba
+
+
+def example_shape(radius, x, y, z):
+    from mitsuba.core.xml import load_string
+
+    xml = f"""
+    <shape version='0.1.0' type="sphere">
+        <float name="radius" value="{radius}"/>
+        <transform name="to_world">
+            <translate x="{x}" y="{y}" z="{z}"/>
+        </transform>
+        <sensor type="irradiancemeter">
+            <film type="hdrfilm">
+                <integer name="width" value="1"/>
+                <integer name="height" value="1"/>
+            </film>
+        </sensor>
+    </shape>
+    """
+    return load_string(xml)
+
+
+def test_construct(variant_scalar_rgb):
+    """
+    We construct an irradiance meter attached to a sphere and assert that the
+    following parameters get set correctly:
+    - to_world transform
+    - associated shape
+    - film
+    """
+    radius = 45.2328
+    x = 2
+    y = 5
+    z = 8.3
+    sphere = example_shape(radius, x, y, z)
+    sensor = sphere.sensor()
+
+    assert sensor.shape() == sphere
+    assert ek.allclose(sensor.film().size(), [1, 1])
+
+
+def test_sampling(variant_scalar_rgb):
+    """
+    We construct an irradiance meter attached to a sphere and assert that sampled
+    rays originate at the sphere's surface
+    """
+    from mitsuba.core import Vector3f
+
+    radius = 45.2328
+    x = 2
+    y = 5
+    z = 8.3
+    sphere = example_shape(radius, x, y, z)
+    sensor = sphere.sensor()
+
+    for i in range(100):
+        r = [random.random() for x in range(6)]
+        ray = sensor.sample_ray_differential(r[0], r[1], [r[2], r[3]], [r[4], r[5]])[0]
+
+        o_rel = Vector3f(ray.o.x - x, ray.o.y - y, ray.o.z - z)
+
+        d_rel = Vector3f(o_rel + ray.d)
+
+        # assert that the ray starts at the sphere surface
+        assert ek.abs(ek.norm(o_rel) - radius) < 1e-5
+
+        # assert that all rays point away from the sphere center
+        ek.norm(d_rel) > ek.norm(o_rel)
+
+
+def test_incoming_flux(variant_scalar_rgb):
```

Comment:
This is a great test!

Could we have the same test but this time using an actual integrator, and calling the `scene.integrator().render()`?

Reply:
I deliberately designed this test without an integrator, because I wanted it to depend on as little other parts of Mitsuba as possible.
I added another test case that does the same but uses the path tracer.